src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Cipher.java

```diff
@@ -799,7 +806,6 @@ private int implDoFinal(byte[] out, int outOfs, int outLen)
             }
             return k;
         } catch (PKCS11Exception e) {
-            doCancel = false;
```

Comment:
Comment for line 638 apply here as well.

Reply:
I agree, but the comment (in my opinion) should be right after the C_EncryptUpdate/C_DecryptUpdate call when doCancel is set to false.